### DISCUSSION TITLE
Possible bug on query[_one] method?

### DISCUSSION THREAD

**Discussion by mindset-team** [Textualize/textual#1041] [General]
There's seems to be some case sensitivity issue with query/query_one. Here's an example app where widget id *must* be lower-cased in order to return a valid query result (vs. a `NoMatches` exception). The query selector seems to be case-insensitive.  

```
from textual.app import App
from textual.widgets import Static

class GridLayoutExample(App):
    def compose(self):
        yield Static("some text here..", id="One")

    async def on_key(self, event):
        print(f'{self.query_one("#One")=}')

app = GridLayoutExample()

if __name__ == "__main__":
    app.run()
```
Above, `on_key` is simply used to trigger the query event and `textual console` opened in another terminal to monitor print output.
The code above fails with `NoMatches: No nodes match <DOMQuery GridLayoutExample(title='GridLayoutExample') filter='#One'`

However, changing `id=One` to `id=one` (in compose method yielding the static widget) works, no exception raised and console printout is valid:

```
[20:58:32] PRINT                                                                                                           latout.py:13
self.query_one("#One")=Static(id='one')
```
Is this behavior expected? It seems counter-intuitive that a selector would be case insensitive, but id would be coerced to be lower-cased for a query to work. I couldn't find any reference to case-[in]sensitivity in the textualize.io guide, or on github (after a _quick_ review of the textual css parsing code and def query). 

Not sure this elevates to a bug (and anyway maintaining all ID's in lower-case would avoid this issue) but since many of the examples have `#One, #Two` identifiers throughout the docs, the issue seems to merit a discussion here, a clear instruction in the docs, or designating as a bug to fix. Thoughts?

**Comment by willmcgugan** [comment]
https://github.com/Textualize/textual/issues/1047

**Comment by davep** [comment]
Just to update, after first implementing the fully case-insensitive approach, we've decided that it'll be better to follow the Web here and go the other way, and enforce case-sensitive.

**Comment by davep** [comment]
Full case-sensitivity now implemented via #1077.
